.github/workflows/upload-release-builds.yml

```diff
@@ -0,0 +1,48 @@
+name: Upload Release Builds
+
+
+# When a release occurs, download artifacts published from build.yml,
+# append the release tag to their names, and upload them to the release.
+
+on:
+  release:
+    types: [published]
+  workflow_dispatch:
+
+jobs:
+  upload-release-builds:
+    name: Upload release builds
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write # release changes require contents write
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Download artifacts from commit ${{ github.sha }}
+        uses: dawidd6/action-download-artifact@v3
+        with:
+          workflow: build.yml
+          workflow_conclusion: success
+          commit: ${{ github.sha }}
+          skip_unpack: true
+          path: dist
+
+      - name: Print dist directory
+        run: ls -R dist
+
+      - name: Upload release builds
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        # We want to append the release tag to the name before upload
+        run: |
+          for file in ${{ github.workspace }}/dist/*; do
+            if [ -f "$file" ]; then
+              FILE_NAME=$(basename "$file")
+              FILE_NAME_NO_EXTENSION=$(basename "$file" .zip)
+
+              NEW_FILE_NAME="${FILE_NAME_NO_EXTENSION%}-OpenAssetIO${{ github.event.release.tag_name }}"
```

Comment:
[fix] Could we start the file name with `OpenAssetIO-v1.0.0` rather than have it at the end?

Reply:
Sure, done as part of the main restructure commit, (I object to the use of [fix] here though bud, this wasn't a mistake, the end is as good as the beginning as far as I'm concerned.)

Reply:
Can we add the hyphen separating the version from OpenAssetIO as well?

As an aside, I went through all the ASWF GitHub org's repos to see how they do artefact naming. Only two repos actually publish any release artefacts, so don't really give enough data points to judge "standard" naming conventions.  But if it's of interest, [MaterialX](https://github.com/AcademySoftwareFoundation/MaterialX/releases) and [OpenCue](https://github.com/AcademySoftwareFoundation/OpenCue/releases) have release artefacts, and have rather different naming schemes.

Reply:
Sure. https://github.com/OpenAssetIO/OpenAssetIO/pull/1241/commits/bf42b54740054871447431bd9cce7484eaa7f4e1